libs/feature-workspaces/src/lib/projects/projects.component.ts

```diff
@@ -60,8 +75,17 @@ export class ProjectsComponent implements OnInit {
       }),
       map((r: any) => {
         const w = r.data.workspace;
-        const projects = w.projects.map((p: any) => {
-          return { ...p, actions: this.createActions(p) };
+        const workspaceSettings = this.settings.getWorkspace(
+          this.workspacePath
+        );
+        this.pinnedProjectNames =
```

Comment:
Should you maybe use a set instead of list? Or are you using the ordering of the list for presentation?

Reply:
Also, did you consider making pinnedProjectNames an observable? Just wondering (I usually find observables to be more reliable with Angular's change detection)